sky/adaptors/ibm.py

```diff
@@ -0,0 +1,42 @@
+
+
+
+from functools import wraps
+from sky.clouds import ibm
+import os
+import yaml
+
+
+ibm_cloud_sdk_core=None
+ibm_vpc=None
+client = None
+
+def import_package(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        global ibm_cloud_sdk_core, ibm_vpc
+        if ibm_cloud_sdk_core is None or ibm_vpc is None:
+            try:
+                import ibm_cloud_sdk_core as _ibm_cloud_sdk_core
+                import ibm_vpc as _ibm_vpc
+                ibm_cloud_sdk_core = _ibm_cloud_sdk_core
+                ibm_vpc = _ibm_vpc
+            except ImportError:
+                raise ImportError('Fail to import dependencies for IBM.'
+                                  'Try pip install "skypilot[ibm]"') from None
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@import_package
+def client(**kwargs):
+    """returns ibm vpc client"""
+
+    if not client:
+        if not os.path.isfile(os.path.expanduser("~/.ibm/credentials")):
```

Comment:
@Cohen-J-Omer how this credentials file looks like? Who creates it? Is there template for it?

Reply:
Similarly to the other providers, the user will initially (before interacting with skypilot) have to run some form of authentication via the cli (see check_credentials() in the cloud providers files, e.g. `aws configure`). 
As usual, our authentication process will have to be more complex, as we'll have to create a script that gathers the required info and outputs it to the aforementioned location.